easypost/services/shipment_service.py

```diff
@@ -54,12 +57,23 @@ def get_next_page(
         optional_params: Optional[Dict[str, Any]] = None,
     ) -> Dict[str, Any]:
         """Get next page of shipment collection."""
-        optional_params = {
-            "include_children": shipments.get("include_children"),
-            "purchased": shipments.get("purchased"),
+        self._check_has_next_page(collection=shipments)
+
+        params = {
+            "before_id": shipments["shipments"][-1].id,
+            "page_size": page_size,
+            "include_children": shipments.get(_FILTERS_KEY, {}).get(
+                "include_children", None
+            ),  # Use the same include_children as the last page
+            "purchased": shipments.get(_FILTERS_KEY, {}).get(
+                "purchased", None
```

Comment:
```suggestion
                "include_children"
            ),  # Use the same include_children as the last page
            "purchased": shipments.get(_FILTERS_KEY, {}).get(
                "purchased"
```
`None` is already the default, let's not include it

Reply:
Resolved